api/python/provisioner/commands/sw_rollback.py

```diff
@@ -0,0 +1,143 @@
+
+    target_version: str = attr.ib(
+        metadata={
+            inputs.METADATA_ARGPARSER: {
+                'help': "CORTX version to rollback "
+                "in case yum rollback is not performed"
+            }
+        },
+        default=None
+    )
+    targets: str = RunArgs.targets
+
+
+@attr.s(auto_attribs=True)
+class SWRollback(CommandParserFillerMixin):
+
+    '''Defines CORTX software rollback logic'''
+
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+    _run_args_type = RunArgsSWRollback
+
+    @staticmethod
+    def _rollback_component(component, targets):
+        state_name = "components.{}.rollback".format(component)
+        try:
+            logger.debug(
+                "Restoring {} configuration on {}".format(component, targets)
+            )
+            StatesApplier.apply([state_name], targets)
+        except Exception:
+            logger.exception(
+                "Failed to restore {} on {}".format(component, targets)
+            )
+            raise
+
+    def run(self, target_version=None, targets=ALL_MINIONS):
+
+        logger.info(f'Starting rollback process on {targets}')
+        local_minion = local_minion_id()
+
+        upgrade_path = PillarKey('upgrade')
+        upgrade_dict = PillarResolver(local_minion).get([upgrade_path])
+        upgrade_dict = upgrade_dict[local_minion][upgrade_path]
+
+        try:
+            if target_version:
+                yum_snapshots = upgrade_dict.get('yum_snapshots', {})
+
+                if not yum_snapshots:
+                    raise ValueError(
+                        'yum snapshots data not available for rollback'
+                    )
+
+                if target_version not in yum_snapshots:
+                    raise ValueError(
+                        f'yum snapshots not available for {target_version}'
+                    )
+                else:
+                    yum_txn_ids = yum_snapshots.get(target_version, {})
+                    logger.debug(
+                        'YumRollbackManager will process yum_txn_ids: '
+                        f'{yum_txn_ids}'
+                    )
+                    for target in yum_txn_ids:
+                        txn_id = (yum_txn_ids[target])
+                        if not txn_id or txn_id is values.MISSED:
+                            raise BadPillarDataError(
+                                f"yum txn id not available for {target}"
+                            )
+                        else:
+                            YumRollbackManager()._yum_rollback(txn_id, target)
+
+            # TODO Add provisioner to sw_list and
+            # reconfigure provisioner through rollback state
+            logger.debug(f'Restoring provisioner configuration on {targets}')
+            config_salt_master()
+
+            config_salt_minions()
+
+            _apply_provisioner_config(targets)
+
+            sw_list = upgrade_dict.get('sw_list', [])
```

Comment:
please log a warning if the list is empty

Reply:
Addressed